File: docs-site/src/snippets/csharp/cache/disk_cache.cs
using Xberg;
using System;
using System.Threading.Tasks;

var config = new ExtractionConfig
{
    UseCache = true,
    CacheNamespace = "xberg_cache",
    CacheTtlSecs = 86400 * 7,
};

Console.WriteLine("First extraction (will be cached)...");
var result1 = (await XbergConverter.ExtractAsync(ExtractInput.FromUri("document.pdf"), config)).Results[0];
Console.WriteLine($"  - Content length: {result1.Content.Length}");

Console.WriteLine("\nSecond extraction (from cache)...");
var result2 = (await XbergConverter.ExtractAsync(ExtractInput.FromUri("document.pdf"), config)).Results[0];
Console.WriteLine($"  - Content length: {result2.Content.Length}");

Console.WriteLine($"\nResults are identical: {result1.Content == result2.Content}");

// NOTE: The C# binding exposes no public API to clear the on-disk cache or
// read cache statistics (no ClearCacheAsync/ClearAllCacheAsync/GetCacheStatsAsync).
// CacheStats.FromJson exists only to parse a JSON string you already have;
// there is no method that produces one. Manage the cache directory directly
// on disk if you need to clear it.

File: docs-site/src/snippets/python/cache/disk_cache.py
import asyncio
from xberg import ExtractInput, extract, ExtractionConfig

config = ExtractionConfig(
    use_cache=True,
    cache_namespace="docs-example",
    cache_ttl_secs=7 * 86400,
)

async def main() -> None:
    print("First extraction (will be cached)...")
    result1 = await extract(ExtractInput(uri="document.pdf"), config)
    print(f"  - Content length: {len(result1.results[0].content)}")

    print("\nSecond extraction (served from cache)...")
    result2 = await extract(ExtractInput(uri="document.pdf"), config)
    print(f"  - Content length: {len(result2.results[0].content)}")

    print(f"\nResults are identical: {result1.results[0].content == result2.results[0].content}")

asyncio.run(main())
